xbmc/FileItem.cpp

```diff
@@ -668,6 +668,10 @@ void CFileItem::Serialize(CVariant& value) const
 
   if (m_gameInfoTag)
     (*m_gameInfoTag).Serialize(value["gameInfoTag"]);
+
+  if (!m_mapProperties.empty())
+    for (const auto& it : m_mapProperties)
+      value["customproperties"][it.first] = it.second;
```

Comment:
It would probably be more efficient to first get a reference to `value["customproperties"]` and use that instead of having to implicitely look it up for every property.

Reply:
Could you offer guidance on that? I'm apparently not quite familiar enough with C++ to get my head around it 😅

Reply:
```cpp
auto& customProperties = values["customproperties"];
for (const auto& prop : m_mapProperties)
  customProperties[prop.first] = prop.second;
```

Reply:
Using that code, I got the following errors on build:
```python
"C:\Users\finkleandeinhorn\Documents\GitHub\xbmc\kodi-build.x64\ALL_BUILD.vcxproj" (default target) (1) ->
"C:\Users\finkleandeinhorn\Documents\GitHub\xbmc\kodi-build.x64\kodi.vcxproj" (default target) (12) ->
"C:\Users\finkleandeinhorn\Documents\GitHub\xbmc\kodi-build.x64\libkodi.vcxproj" (default target) (14) ->
(ClCompile target) ->
  c:\users\finkleandeinhorn\documents\github\xbmc\xbmc\fileitem.cpp(675): error C2065: 'customProperties': undeclared identifier [
C:\Users\finkleandeinhorn\Documents\GitHub\xbmc\kodi-build.x64\libkodi.vcxproj]
```
Which is especially confusing to me, because it *has* been declared 🤔

Reply:
`auto& customProperties = value["customproperties"];`
value not values

Reply:
Yes, I caught that change myself. The error above is something else.

Should it maybe be `const auto& customProperties`?

Reply:
No `const` is wrong, you are changing the value. 
```
  auto& customProperties = value["customproperties"];
  for (const auto& prop : m_mapProperties)
    customProperties[prop.first] = prop.second;
```
complies for me (on VS2017). Commit the change and we will see what jenkins does.